route-rs-runtime/src/link/blackhole_link.rs

```diff
@@ -0,0 +1,128 @@
+use crate::link::PacketStream;
+use futures::{Async, Future, Poll};
+
+pub struct BlackHoleLink<Packet: Sized> {
+    input_stream: PacketStream<Packet>,
+}
+
+impl<Packet: Sized> BlackHoleLink<Packet> {
+    pub fn new(input_stream: PacketStream<Packet>) -> Self {
+        BlackHoleLink { input_stream }
+    }
+}
+
+impl<Packet: Sized> Future for BlackHoleLink<Packet> {
+    type Item = ();
+    type Error = ();
+
+    /*
+    4 cases: Async::Ready(Some), Async::Ready(None), Async::NotReady, Err
```

Comment:
Use `///` so they are doc comments

Reply:
Since we're familiar with the `poll` contract now, these comments probably aren't necessary anymore.

Reply:
Going to drop this explanation. Will leave the original up in async_link, will pull into documentation eventually.